kinto/core/views/heartbeat.py

```diff
@@ -10,10 +12,11 @@
 def get_heartbeat(request):
     """Return information about server health."""
     status = {}
-
     heartbeats = request.registry.heartbeats
-    for name, callable in heartbeats.items():
-        status[name] = callable(request)
+    seconds = float(request.registry.settings['heartbeat_timeout_seconds'])
+    with timeout(seconds):
```

Comment:
I would probably go for a global timeout as well as a timeout per heartbeat too.